tests/lib/rules/no-jquery-methods.js

```diff
@@ -0,0 +1,100 @@
+const rule = require('../../../lib/rules/no-jquery-methods');
+const MESSAGE = rule.meta.message;
+const BLACKLISTMETHOD = rule.meta.blacklist[0];
+const RuleTester = require('eslint').RuleTester;
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module'
+  }
+});
+
+ruleTester.run('no-jquery-methods', rule, {
+  valid: [
+    {
+      code: `
+        export default Ember.Component({
+          init() {
+            Ember.$().notBlacklistedMethod();
+          }
+        });`
+    },
+    {
+      code: `
+        export default Ember.Component({
+          init() {
+            this.$().notBlacklistedMethod();
+          }
+        });`
+    },
+    {
+      code: `
+        export default Ember.Component({
+          init() {
+            const myJQueryObj = this.$();
+            myJQueryObj[${BLACKLISTMETHOD}]();      
```

Comment:
My original point was that this should be invalid, but detecting it would be very difficult as you'll have to walk back the reference or keep track of all variables bound to jquery values